src/components/_Controllers/CMSPreview.js

```diff
@@ -35,6 +37,9 @@ class CMSPreview extends Component {
     //TODO: one endpoint for revisions data requests
     let req;
     switch (page_type) {
```

Comment:
do you think we could get away without this switch somehow so we don't have to keep adding to it for new content-types?

Reply:
I'm not sure, the way we generate the endpoints requires object types that will end up using different queries.
```
query getPageRevision($id: ID) {
  pageRevision(id: $id) {
    id
    asServicePage {
```

Unless we do some extensive refactoring on the backend our revisions will continue to be different objects types in graphQL and require different queries